### PR TITLE
config: use lower default RAM hint for openstack provisioning

### PR DESCRIPTION
**Pull request by smithfarm**
Fixes: https://trello.com/c/X5StTYxZ/378-teuthology-runs-in-ecp-use-needlessly-large-flavor
Related-to: https://github.com/SUSE/ceph/pull/253
Signed-off-by: Nathan Cutler <ncutler@suse.com>

---

- [ ] https://github.com/SUSE/ceph/pull/253 merged
- [ ] https://github.com/SUSE/ceph/pull/253 backported to ses5

### CODE REVIEW

**Comment by smithfarm** [comment]
Note: 4GB of RAM will be too little for Bluestore OSDs, which might use up to 4GB of RAM per OSD in the default configuration.

File: teuthology/openstack/__init__.py [hunk 1]
```diff
@@ -84,6 +84,7 @@ def __init__(self, name_or_id, info=None):
             errmsg = 'VM creation failed'
             if 'message' in self.info:
                 errmsg = '{}: {}'.format(errmsg, self.info['message'])
+            OpenStack().run("server delete " + self.info.get('id', self.name_or_id))
```
Comment (kshtsk) [inline teuthology/openstack/__init__.py]: This commit is not related to the PR, the "server delete" should not be run directly from the constructor of this class because it does not run "server create", in other words this instruction should be run upper in the stack call.

Comment (smithfarm) [reply teuthology/openstack/__init__.py]: dropped the commit

**Comment by smithfarm** [comment]
@kshtsk Please take another look.

**Comment by smithfarm** [comment]
Using 8GB RAM and 4 VCPUs per test machine is really excessive. If we ever have a test that needs bigger machines, we can include the necessary `openstack:` -> `machine:` hint in the YAML of that test.

**Comment by kshtsk** [comment]
@susebot run teuthology for suse:tier0

**Comment by kshtsk** [comment]
From the log:
```2018-12-19T22:33:43.921 DEBUG:teuthology.misc::sh: flock --close --timeout 28800 /tmp/teuthology-server-create.lock openstack --quiet server create  -f json  --image 'teuthology-sle-15.0-x86_64' --flavor 'm1.medium' --key-name teuth-48  --user-data /tmp/tmpXXOs6o --nic net-id=1132724b-a83c-4f07-b4d6-e701fd3ebf46 --min 1 --max 1 --security-group teuth-48-worker --property teuthology=da4d49bb6c0324e32d86f5724b32f90b --property ownedby=192.168.0.16 --wait  target
```

Where flavor `m1.medium` is: ```{u'Name': u'm1.medium', u'RAM': 4096, u'Ephemeral': 0, u'VCPUs': 2, u'Is Public': True, u'Disk': 40, u'ID': u'3'}``` which is 4GB RAM, I guess it meets expectation
